From b1b7ece6ca12c1eea14e9a6dc69ec138f6298c9e Mon Sep 17 00:00:00 2001
From: Pauli Nieminen <suokkos@gmail.com>
Date: Fri, 8 Nov 2013 08:11:26 +0200
Subject: [PATCH] Avoid sending package name with arch to status FD

Sending arch with package name to status fd makes other side parse the
string incorrectly because expectation is to do simple string split with
colon. But of course there would be an extra colon in the string.

Signed-off-by: Pauli Nieminen <suokkos@gmail.com>
---
 apt-pkg/deb/dpkgpm.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/apt-pkg/deb/dpkgpm.cc b/apt-pkg/deb/dpkgpm.cc
index 26d79db..92b85b6 100644
--- a/apt-pkg/deb/dpkgpm.cc
+++ b/apt-pkg/deb/dpkgpm.cc
@@ -664,7 +664,7 @@ void pkgDPkgPM::ProcessDpkgStatusLine(char *line)
       }
       std::string msg;
       strprintf(msg, _(iter->second), i18n_pkgname.c_str());
-      d->progress->StatusChanged(pkgname, PackagesDone, PackagesTotal, msg);
+      d->progress->StatusChanged(short_pkgname, PackagesDone, PackagesTotal, msg);
 
       // FIXME: this needs a muliarch testcase
       // FIXME2: is "pkgname" here reliable with dpkg only sending us 
@@ -693,7 +693,7 @@ void pkgDPkgPM::ProcessDpkgStatusLine(char *line)
          PackagesDone++;
 
          strprintf(msg, translation, i18n_pkgname.c_str());
-         d->progress->StatusChanged(pkgname, PackagesDone, PackagesTotal, msg);
+         d->progress->StatusChanged(short_pkgname, PackagesDone, PackagesTotal, msg);
          
       }
       if (Debug == true) 
-- 
1.8.4.2

